frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -50,88 +58,132 @@ const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updat
-                  text={ComponentText.CANCEL}
-                />
-                <Button
-                  type="button"
-                  size="sm"
-                  disabled={false}
-                  text={ComponentText.ADD}
-                  handleClick={() => addFarmInfo(values)}
-                />
-              </StyledButtonGroupContainer>
-            </StyledTextAreaContainer>
-            <StyledListContainer>
-              <StyledListItem>
-                <h4>Field Name</h4>
-                <p>{fieldsInfo.FieldName}</p>
-              </StyledListItem>
-              <StyledListItem>
-                <h4>Area</h4>
-                <p>{fieldsInfo.Area}</p>
-              </StyledListItem>
-              <StyledListItem>
-                <h4>Comments</h4>
-                <p>{fieldsInfo.Comments}</p>
+    <>
+      {isSubmitted ? (
+        <StyledFieldInfoList>
+          <StyledListContainer>
+            <StyledListItem width="20%">
+              <h4>Field Name</h4>
+              <p>{fieldsInfo.Fields[fieldIndex].FieldName}</p>
+            </StyledListItem>
+            <StyledListItem width="20%">
+              <h4>Area</h4>
+              <p>{fieldsInfo.Fields[fieldIndex].Area}</p>
+            </StyledListItem>
+            <StyledCommentContainerDesktop>
+              <StyledListItem width="100%">
+                <h4>Comments (optional)</h4>
+                <p>{fieldsInfo.Fields[fieldIndex].Comments}</p>
               </StyledListItem>
-            </StyledListContainer>
-          </StyledFarmInfo>
-        </Form>
-      )}
-    </Formik>
+            </StyledCommentContainerDesktop>
+            <StyledFontAwesomeContainer>
+              <FontAwesomeIcon icon={faPencil} />
+              <FontAwesomeIcon icon={faTrash} />
+            </StyledFontAwesomeContainer>
+          </StyledListContainer>
+          <StyledCommentContainerMobile>
+            <StyledListItem width="100%">
+              <h4>Field Comments (optional)</h4>
+              <p>{fieldsInfo.Fields[fieldIndex].Comments}</p>
+            </StyledListItem>
+          </StyledCommentContainerMobile>
+          {!fieldAdd ? (
+            <Button
+              type="button"
+              size="md"
+              disabled={false}
+              radius="50px"
+              actions="secondary"
+              text={ComponentText.NEWFIELD}
+              handleClick={addNewField}
+            >
+              <FontAwesomeIcon icon={faPlus} />
+            </Button>
+          ) : null}
+        </StyledFieldInfoList>
+      ) : null}
+      {fieldAdd || !isSubmitted ? (
+        <Formik
+          initialValues={initialValues}
+          validationSchema={validationSchema}
+          onSubmit={onSubmit}
+        >
+          {({ values }) => (
+            <Form>
+              <StyledFarmInfo>
+                <div id="inputContainer">
+                  <CustomField
+                    label="Field name"
+                    id="FieldName"
```

Comment:
In react you don't really need ID tags since you have tighter control over everything when using React. Even still, semantically it could be camelCase over PascalCase https://stackoverflow.com/questions/51554743/avoid-using-uppercase-in-class-and-id-in-css